src/libs/actions/Report.js

```diff
@@ -1011,12 +1012,48 @@ function fetchAllReports(
 
             // Get all the chat reports if they have any, otherwise create one with concierge
             if (reportIDs.length > 0) {
-                return fetchChatReportsByIDs(reportIDs);
+                // return fetchChatReportsByIDs(reportIDs);
+                const simplifiedReports = {};
+                return API.GetChats({reportIDList: reportIDs})
+                    .then((chats) => {
+                        // Process the reports and store them in Onyx. At the same time we'll save the simplified reports in this
+                        // variable called simplifiedReports which hold the participants (minus the current user) for each report.
+                        // Using this simplifiedReport we can call PersonalDetails.getFromReportParticipants to get the
+                        // personal details of all the participants and even link up their avatars to report icons.
+                        const reportIOUData = {};
+                        _.each(chats.reportList, (report) => {
+                            const simplifiedReport = getSimplifiedReportObject(report);
```

Comment:
I assume removing getSimplifiedReportObject (and massaging the data in auth/php) is out of scope for this PR?

Reply:
Yeah
- getSimplifiedReportObject is used in other places
- [PersonalDetails->getDisplayName](https://github.com/Expensify/Web-Expensify/pull/33602/files#diff-cb404322b33af06216873817c1583a864f2ec874002870f9571d9a07353eb919R274) -- the personal details logic could start getting implemented in auth, but I think it's better to keep that out of the scope of this because it would create duplication in PHP and Auth, and this change will be big enough already
- [Report_Action_IOU->getMessage](https://github.com/Expensify/Web-Expensify/pull/33602/files#diff-cb404322b33af06216873817c1583a864f2ec874002870f9571d9a07353eb919R292), this one is actually one that you pointed out, and if you look at the code there's a lot in there that would need to be moved or duplicated in auth, so leaving that out of scope too

Reply:
ok makes sense, so we'll need more refactoring after this one 👍